sfge/src/main/java/com/salesforce/rules/unusedmethod/RuleStateTracker.java

```diff
@@ -238,6 +238,63 @@ boolean classInheritsMatchingMethod(String definingType, String signature) {
         return MethodUtil.getMethodWithSignature(g, definingType, signature, true).isPresent();
     }
 
+    /**
+     * Get all {@link NewObjectExpressionVertex} instances representing instantiations of an object
+     * whose type is {@code constructedType}.
+     */
+    List<NewObjectExpressionVertex> getConstructionsOfType(String constructedType) {
+        // Start off with all NewObjectExpressionVertex instances whose declared type references the
+        // desired type.
+        List<NewObjectExpressionVertex> results =
+                SFVertexFactory.loadVertices(
+                        g,
+                        g.V()
+                                .where(
+                                        H.has(
+                                                NodeType.NEW_OBJECT_EXPRESSION,
+                                                Schema.TYPE,
+                                                constructedType)));
+
+        // Inner types can be referenced by outer/sibling types by just the inner name rather than
+        // the full name.
+        // This means we need to do more, but what that "more" is depends on whether we're looking
+        // at an inner or outer type.
+        boolean typeIsInner = constructedType.contains(".");
+        if (typeIsInner) {
```

Comment:
The contents of this `if` branch are very similar to those of the one at line 316, but I couldn't see a way to cleanly refactor them into a single method. In the fullness of time, we may want to revisit that.

Reply:
How about extracting a method and passing in `NodeType.<vertex_type>`, inner class name, and outer class name as input?
The name-splitting could become a util method in `ClassUtil`.